dev/ProgressBar/ProgressBar.cpp

```diff
@@ -93,32 +94,47 @@ void ProgressBar::UpdateStates()
     }
     else if (!IsIndeterminate())
     {
-        SetProgressBarIndicatorWidth();
         winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
     }
 }
 
 void ProgressBar::SetProgressBarIndicatorWidth()
 {
+    const auto templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+
     if (auto&& progressBar = m_layoutRoot.get())
     {
         if (auto&& progressBarIndicator = m_progressBarIndicator.get())
         {
             const double progressBarWidth = progressBar.ActualWidth();
+            const double prevIndicatorWidth = progressBarIndicator.ActualWidth();
             const double maximum = Maximum();
             const double minimum = Minimum();
             const auto padding = Padding();
 
-            if (std::abs(maximum - minimum) > DBL_EPSILON)
+            m_isUpdating = true; // Adds "Updating" state in between to trigger RepositionThemeAnimation Visual Transition in ProgressBar.xaml when reverting back to previous state
```

Comment:
Interesting approach. If you just want a Storyboard to run an alternative approach is to just add the Storyboard as a resource to the root of the ControlTemplate. Then you can get that Storyboard by name and play it as needed. However, this approach seems like a reasonable way to implement the animation.

Reply:
I referenced the old implementation of progressBar with the `Updating` state.